_posts/2023-06-28-#27307.md

```diff
@@ -7,18 +7,86 @@ authors: [ishaanam]
 components: ["wallet"]
 host: ishaanam
 status: upcoming
-commit:
+commit: 0538ad7
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+- In Bitcoin Core, every [wallet transaction](https://github.com/bitcoin/bitcoin/blob/7f0b79ea132d22ad5212c1d3ff4325715ca5ac12/src/wallet/transaction.h#L160)
+  has a [transaction state](https://github.com/bitcoin/bitcoin/blob/7f0b79ea132d22ad5212c1d3ff4325715ca5ac12/src/wallet/transaction.h#L68) (detailed [here](https://gist.github.com/ishaanam/846adf3b453c3a85fe6e15c882c57ae0#types-of-tx-states)).
+  These transaction states are part of how the wallet decides which
+  transactions to allow the user to spend, and which transactions to
+  count towards a user's balance.
+
+- Wallet transaction states and conflicts were previously discussed in
+  review club [#27145](/27145)
+
+- When a transaction is `TxStateInactive`:
+    - its inputs are considered spent, and
+    - its outputs are not available to spend and don't count towards a
+      user's balance
+
+- When a transaction is `TxStateConflicted`:
+    - its inputs are _not_ considered spent, and
+    - its outputs are not available to spend and don't count towards a
+      user's balance
+
+- On master, wallet txs are only considered conflicted when the conflicting tx
+  gets mined into a block. This means that if a transaction is only conflicted
+  by a mempool tx, it is considered `TxStateInactive` instead. This can can lead to
+  [confusion](https://bitcoin-irc.chaincode.com/bitcoin-core-dev/2023-05-09#1683605157-1683612219;)
```

Comment:
```suggestion
  [confusion](https://bitcoin-irc.chaincode.com/bitcoin-core-dev/2023-05-09#1683605157-1683612219)
```
(but doesn't seem to do any harm)

Reply:
The `;` is actually part of the URL as produced by the server, even though indeed it doesn't matter.